### PR TITLE
Fix bug 1255341. Fixed sorting of the results in the machinery.

### PR DESCRIPTION
**Pull request by jotes**
hi! @mathjazz Could you check if this patch fixed problems mentioned by @MikkCZ ?

### CODE REVIEW

**Comment by mathjazz** [comment]
I think what @MikkCZ wants is that you sort sources for a suggestions from multiple sources in the same order as we sort suggestions.

E.g.: 
`100% • Translation memory9 • Open Source • Mozilla`
should become
`100% • Translation memory9 • Mozilla • Open Source`

File: pontoon/base/static/js/main.js [hunk 1]
```diff
@@ -133,7 +133,7 @@ var Pontoon = (function (my) {
         var title = loader !== 'search' ? ' title="Copy Into Translation (Tab)"' : '',
             sources = sourcesMap[data.original + data.translation];
 
-        if (sources) {
+        if ($.type(sources) !== 'undefined') {
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Why not `typeof`?

**Comment by jotes** [comment]
@mathjazz strangely, previous code didn't work in some cases for the order of single source results. E.g. after i set breakpoint in the append(). That probably has misled me and I didn't even ask for the screenshot/some more informations. However, i can fix the order of items in results with multiple sources. Give me a moment!

**Comment by jotes** [comment]
@mathjazz Ok, I added sorting for the multi-source items.

File: pontoon/base/static/js/main.js [hunk 2]
```diff
@@ -185,11 +185,29 @@ var Pontoon = (function (my) {
...
               sourceA = getTranslationSource(a),
               sourceB = getTranslationSource(b);
-
           return (valA < valB) ? 1 : (valA > valB) ? -1 : (sourceA > sourceB) ? 1 : (sourceA < sourceB) ? -1 : 0;
         });
-        ul.append(listitems);
-
+        ul.html(listitems);
+
+        // Sort sources inside results.
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: No period at the end of a single sentence comment

File: pontoon/base/static/js/main.js [hunk 3]
```diff
@@ -185,11 +185,29 @@ var Pontoon = (function (my) {
...
+        ul.find('header').each(function (index) {
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Remove space in front of `(index)`.

File: pontoon/base/static/js/main.js [hunk 4]
```diff
@@ -185,11 +185,29 @@ var Pontoon = (function (my) {
...
+          var $header = $(this),
+              sources = $header.children('a'),
+              sortedItems = sources.sort(function(a, b) {
+                var sourceA = sourceMap[$(a).children('span').text()],
+                    sourceB = sourceMap[$(b).children('span').text()];
+                return (sourceA > sourceB) ? 1 : (sourceA < sourceB) ? -1 : 0;
+              });
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Add blank line.

File: pontoon/base/static/js/main.js [hunk 5]
```diff
@@ -185,11 +185,29 @@ var Pontoon = (function (my) {
...
+           // Remove all &bull entities
+           $header.contents().filter(function() {
+              return this.nodeType === 3;
+           }).remove();
+           $header.children('a').remove();
+           sortedItems.each(function() {
+              $header.append('&bull;');
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Add space: `' &bull; '`

**Comment by mathjazz** [comment]
We probably shouldn't add any &bull; when composing HTML and then we also don't need to remove them. Let's just add them at the end where it makes sense.

**Comment by jotes** [comment]
@mathjazz I've migrated sources into <ul><li> structure but i would need your help with the css :(

**Pull request merged**